tools/lisa-build-asset

```diff
@@ -57,25 +73,147 @@ def list_assets(recipe_dir):
+
+def is_alpine_build(asset, arch, native_build, **kwargs):
+    has_native_build = is_property_set(asset, arch, "ALPINE_BUILD_DEPENDENCIES", **kwargs)
+    has_broken_cross_build = is_property_set(asset, arch, "BROKEN_CROSS_COMPILATION", **kwargs)
+
+    if not has_native_build and has_broken_cross_build:
+        raise ValueError(f'\n Asset {asset} cannot be built on Alpine Linux or cross compiled')
 
-def make(asset, actions, arch, toolchain, recipe_dir, build_dir):
+    if native_build:
+        if has_native_build:
+            return True
+        print(f'\n Asset {asset} cannot be built on Alpine Linux, falling back to cross compilation')
+        return False
+    else:
+        if has_broken_cross_build:
+            print(f'\n Asset {asset} cannot be cross-compiled, falling back to Alpine Linux')
+            return True
+        return False
+
+@contextmanager
+def make_chroot(asset, arch, build_dir, recipe_dir, arch_chroot_dir):
+    # Create a chroot at arch_chroot_dir as a contextmanager.
     recipe = get_asset_recipe(asset, recipe_dir)
-    env = get_env(asset, arch, toolchain, recipe, build_dir)
-    for action in actions:
-        print('\nMaking {} {} {} in: {}\n'.format(arch, asset, action, build_dir))
+    env = get_env(
+        asset=asset,
+        arch=arch,
+        recipe=recipe,
+        build_dir=build_dir
+    )
 
-        script = 'source {recipe} && {action}'.format(
-            recipe=recipe,
-            action=action,
+    # List the required Alpine packages.
+    try:
+        alpine_dependencies = get_asset_property(
+            asset, arch, "ALPINE_BUILD_DEPENDENCIES",
+            build_dir=build_dir,
+            recipe_dir=recipe_dir,
+            arch_chroot_dir=arch_chroot_dir
         )
-        cmd = ['bash', '-c', script]
-        subprocess.check_call(cmd, env=env, cwd=build_dir)
+    except ValueError:
+        alpine_dependencies = []
+    else:
+        alpine_dependencies = ('-p', ' '.join(map(quote, alpine_dependencies)))
+
+    # List the required env variables.
+    env_args = ('-k', ' '.join(env.keys())) if env else []
+
+    # Install the chroot.
+    subprocess.check_call(
+        [
+            'sudo', os.path.join(get_alpine_dir(), 'alpine-chroot-install'),
+            '-d', arch_chroot_dir,
+            '-i', LISA_HOME,
+            '-a', QEMU_ARCH_NAMES.get(arch, arch),
+            *alpine_dependencies,
+            *env_args,
+        ],
+        cwd=build_dir,
+    )
 
-ARCHITECTURES = ['arm64', 'armeabi', 'x86_64']
+    def bind_mount(path):
+        # join() will ignore everything before an absolute path so make it
+        # relative first
+        rel = os.path.relpath(path, start='/')
+        path_ = os.path.join(arch_chroot_dir, rel)
+        subprocess.check_call(['sudo', 'mkdir', '-p', path_])
+        subprocess.check_call(['sudo', 'mount', '--bind', path, path_])
+
+    # If build_dir is not in LISA_HOME, bind it aswell.
+    if not build_dir.startswith(LISA_HOME):
```

Comment:
This would incorrectly return `True` for 
```
build_dir='/home/foo/lisa-build'
LISA_HOME='/home/foo/lisa'
```
So we need something like:
```
def is_parent(parent, child):
    parent = os.path.abspath(parent)
    child = os.path.abspath(child)
    rel = os.path.relpath(parent, child)
    # If the relative path only includes ., .. and /, it's a subfolder
    return not rel.replace('.', '').replace('/', '')
```

Reply:
Even though this would be redundant to bind build_dir in this case, this is still working fine if we bind it 2 times. I removed the check